sdk/monitor/monitor-opentelemetry-exporter/src/sampling.ts

```diff
@@ -54,20 +54,43 @@ export class ApplicationInsightsSampler implements Sampler {
     links: Link[],
   ): SamplingResult {
     let isSampledIn = false;
-    if (this._sampleRate === 100) {
-      isSampledIn = true;
-    } else if (this._sampleRate === 0) {
-      isSampledIn = false;
-    } else {
-      isSampledIn = this._getSamplingHashCode(traceId) < this._sampleRate;
-    }
-    // Add sample rate as span attribute
     attributes = attributes || {};
 
+    // Try to get the parent sampling result first
+    const parentSpan = trace.getSpan(context);
+    const parentSpanContext = parentSpan?.spanContext();
+
+    if (
+      parentSpanContext &&
+      trace.isSpanContextValid(parentSpanContext) &&
+      !parentSpanContext.isRemote
+    ) {
+      // If the parent span is valid and not remote, we can use its sample rate
+      const parentSampleRate = Number((parentSpan as any).attributes?.[AzureMonitorSampleRate]);
+      if (!isNaN(parentSampleRate)) {
+        this._sampleRate = Number(parentSampleRate);
+      }
+      if ((parentSpanContext.traceFlags & TraceFlags.SAMPLED) === TraceFlags.SAMPLED) {
+        isSampledIn = true;
+      } else {
+        isSampledIn = false;
+      }
```

Comment:
```suggestion
isSampledIn = (parentSpanContext.traceFlags & TraceFlags.SAMPLED) === TraceFlags.SAMPLED
```
Could reduce nested-if logic here with a simplified assignment.